cuqi/distribution/_distribution.py

```diff
@@ -154,7 +154,7 @@ def logd(self, *args, **kwargs):
             kwargs = self._parse_args_add_to_kwargs(cond_vars, *args, **kwargs)
 
             # Check if all conditioning variables are specified
-            if not all([key in kwargs for key in cond_vars]):
+            if not all([key in kwargs for key in cond_vars]) or len(kwargs) < len(cond_vars)+1:
```

Comment:
I tried to understand the other condition  " len(kwargs) < len(cond_vars)+1"
and I think it is for cases like this below where the first clause is false and the second is true 
`x.logd( s=2)`

In this case, the error msg is "To evaluate the log density all conditioning variables must be specified. Conditioning variables are: ['s']". I thought it will be more informative if the error msg in this particular case mentions x as well?.

Reply:
where:
`x = cuqi.distribution.Gaussian(0, cov=lambda s:s)`

Reply:
I suggest updating the comment to explain the additional condition.